test/tests/functional/pbs_resc_custom_perm.py

```diff
@@ -0,0 +1,164 @@
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Test_Custom_Resource_Perm(TestFunctional):
+    rsc_list = ['foo_str', 'foo_strm', 'foo_strh']
+    hook_list = ["start", "begin"]
+
+    def tearDown(self):
+        self.du.set_pbs_config(confs={'PBS_SERVER': self.server.hostname})
+        TestFunctional.tearDown(self)
+        for rsc in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC,
+                                    id=rsc, runas=ROOT_USER)
+            except PbsManagerError:
+                pass
+        for hook in self.hook_list:
+            self.server.manager(MGR_CMD_DELETE, HOOK, id=hook, runas=ROOT_USER)
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        for rsc in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC,
+                                    id=rsc, runas=ROOT_USER)
+            except PbsManagerError:
+                pass
+
+        self.momA = self.moms.values()[0]
+        self.momA.restart()
+        self.momA.delete_vnode_defs()
+        self.hostA = self.momA.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+        self.server.expect(NODE, {'state': 'free'}, id=self.hostA)
+
+        # Next set custom resources with m flag and without m flag
+        # via qmgr -c 'create resource'
+        attr = {'type': 'string', 'flag': 'h'}
+        r = 'foo_str'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
```

Comment:
The person running PTL is a manager, you don't need to run as root.

Reply:
I will remove it.